File: 13/hs/1.hs
import Data.List.Split (splitOn)
import System.IO
import Utils (compare, mapIdx)

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let pairs = mapIdx (\x i -> (i + 1, x)) $ map lines $ splitOn "\n\n" contents
      right = filter (\(i, [a, b]) -> Utils.compare a b == 1) pairs
      total = sum (map fst right)
  print ("Sum of indices of correct pairs: " ++ show total)
  hClose h

File: 13/js/1.js
const fs = require('fs');
const { compare } = require('./utils');

(function () {
  const contents = fs.readFileSync("../input.txt").toString().trim();
  const pairs = contents.split("\r\n\r\n").map(x => x.trim().split("\r\n")).map((x, i) => ([i + 1, x]));
  const right = pairs.filter(([, [l, r]]) => compare(l, r));
  const sum = right.map(([i,]) => i).reduce((a, b) => a + b, 0);
  console.log("Sum of indices of correct pairs: " + sum);
})();